### PR TITLE
MAILBOX-362 Serialization for MailboxACLUpdated Event

### PR DESCRIPTION
**Pull request by hoangdat**

### CODE REVIEW

File: mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala [hunk 1]
```diff
@@ -38,27 +38,6 @@ private sealed trait Event {
 }
 
 private object DTO {
-
-  object MailboxPath {
```
Comment (mbaechler) [inline mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala]: I don't agree, that's some DTO

Comment (chibenwa) [reply mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala]: I'm the one who proposed this change.

The idea was to only keep the DTO related to Event. keeping other DTO here hurts readability (too much to review) and mixes concerns (mixing DTO of evets and other stuff do not seem good to me).

Maybe we can rename `DTO` to `EventDTOs` to clearly express this concern?

If you are strictly against these extraction to separate files, maybe separating `DTOs` & `EventDTOs` would satisfy us both?

File: mailbox/event/json/src/test/java/org/apache/james/event/json/MailboxACLUpdatedEventSerializationTest.java [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+                    "    \"sessionId\":6," +
+                    "    \"user\":\"user@domain@secondDomain\"" +
+                    "   }" +
+                    "}").get())
+                    .isInstanceOf(IllegalArgumentException.class);
+            }
+        }
+
+        @Nested
+        class DeserializationErrorOnACLDiff {
```
Comment (chibenwa) [inline mailbox/event/json/src/test/java/org/apache/james/event/json/MailboxACLUpdatedEventSerializationTest.java]: I believe that we are missing some tests here:

 - what happens when oldACL is null? is a String? Contains some AclEntryKey that can not be deserialized (empty string? Or `-`)? When rights is null? Not a String? Invalid (like `\"invalid\"`)? Have double right entry `\"rr\"` ?
 - Same thing for new ACL

ACL is a complex object, and we should try to cover the edge cases with tests...

File: mailbox/event/json/src/test/java/org/apache/james/event/json/MailboxACLUpdatedEventSerializationTest.java [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+        private final String NULL_USER = null;
+        private static final String JSON_2 = "{" +
+            "  \"MailboxACLUpdated\":{" +
+            "    \"mailboxPath\":{" +
+            "       \"namespace\":\"#private\"," +
+            "       \"name\":\"mailboxName\"" +
+            "      }," +
+            "    \"aclDiff\":{" +
+            "       \"oldACL\":{\"entries\":{}}," +
+            "       \"newACL\":{\"entries\":{\"$any\":\"ar\"}}}," +
```
Comment (chibenwa) [inline mailbox/event/json/src/test/java/org/apache/james/event/json/MailboxACLUpdatedEventSerializationTest.java]: I don't like that `entries` field... This level of nesting do not add much.

Can we get rid of it and only have `\"newACL\":{\"$any\":\"ar\"}},` ?

Hint you can do this by changing : 

```
case class ACLDiff(oldACL: MailboxACL, newACL: MailboxACL) {...}
```

to 

```

case class ACLDiff(oldACL: Map[JavaMailboxACL.EntryKey, JavaMailboxACL.Rfc4314Rights], 
    newACL: Map[JavaMailboxACL.EntryKey, JavaMailboxACL.Rfc4314Rights]) {...}
```

(And get rid of the Java MailboxACL POJO at the same time ^^

File: mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala (hunk 1 above)
Comment (chibenwa) [reply mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala]: As discussed with @mbaechler , we came to the conclusion having a separated file for DTOs that are not Event was a good idea (but not one file per DTO). Can you group them in one `DTOs.scala` file?

File: mailbox/event/json/src/test/java/org/apache/james/event/json/MailboxACLUpdatedEventSerializationTest.java [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+
+    @Test
+    void mailboxACLUpdatedShouldBeDeserialized() {
+        assertThat(EVENT_SERIALIZER.fromJson(JSON_1).get())
+            .isEqualTo(mailboxACLUpdatedEvent);
+    }
+
+    @Test
+    void mailboxACLUpdatedShouldBeDeserializedFromFile() {
+        assertThat(EVENT_SERIALIZER.fromJson(ClassLoaderUtils
```
[the remark this answers is not in the document]
Comment (hoangdat) [reply mailbox/event/json/src/test/java/org/apache/james/event/json/MailboxACLUpdatedEventSerializationTest.java]: I think it make sure that we can deserialize a json file. It is required in [my ticket](https://ci.linagora.com/linagora/lgs/openpaas/james/issues/861)
> Include JSON loaded from a resource file to demonstrate stability.

**Comment by hoangdat** [comment]
```
[a26f8e93a2a1f2d9d68609e4854fb0a3596a7578] [ERROR] Errors: 
[a26f8e93a2a1f2d9d68609e4854fb0a3596a7578] [ERROR]   MailboxAddedSerializationTest$NullNameSpaceInMailboxPath.mailboxAddedShouldBeWellDeSerializedWhenMissingNameSpace:123 » NoSuchElement
[a26f8e93a2a1f2d9d68609e4854fb0a3596a7578] [ERROR]   MailboxAddedSerializationTest$NullNameSpaceInMailboxPath.mailboxAddedShouldBeWellDeSerializedWhenNullNameSpace:142 » NoSuchElement
```
I don't know why this error ocurred

**Pull request closed without merging**